lisp/ess-r-mode.el

```diff
@@ -1363,11 +1363,12 @@ selected (see `ess-r-set-evaluation-env')."
   :type 'hook
   :group 'ess-R)
 
-(cl-defmethod inferior-ess-reload--override (start-args
+(cl-defmethod inferior-ess-reload--override (start-name start-args
                                              &context ((string= ess-dialect "R") (eql t)))
   "Call `run-ess-r' with START-ARGS.
 Then run `inferior-ess-r-reload-hook'."
-  (run-ess-r start-args)
+  (let ((inferior-ess-r-program start-name))
+    (run-ess-r start-args))
```

Comment:
Can't we use `inferior-ess--start-args` here and avoid needing another argument to the function?

Reply:
Did you mean `--start-name`?

Reply:
If `inferior-ess-reload--override` becomes a public customisation point (i.e. moves to `inferior-ess-reload-override`), it can't use private data. At that point it might make sense to bundle inferior metadata in an alist so we only need to pass a single argument. Until then it's probably best to keep the data explicit.

But I think the program name will eventually be passed as explicit argument to `inferior-ess` rather than through the cust-alist, so this reload signature would mirror the startup signature nicely.

Reply:
Do we really need this overwrite? How about saving the actual runner as a local as well and re-call it?

Reply:
This could probably work, but I prefer to keep this PR small.